lib/main.dart

```diff
@@ -1,24 +1,36 @@
 import 'package:flutter/material.dart';
-import 'package:flutter_localizations/flutter_localizations.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
-import 'package:survey/pages/login_page.dart';
+import 'package:flutter_localizations/flutter_localizations.dart';
+import 'package:get/get.dart';
+import 'package:survey/api/api_client_provider.dart';
+import 'package:survey/navigator/navigator.dart';
+import 'package:survey/pages/home/home_page.dart';
+import 'package:survey/pages/login/login_binding.dart';
+import 'package:survey/pages/login/login_page.dart';
+import 'package:survey/pages/splash/splash_page.dart';
 import 'package:survey/themes.dart';
 
+import 'api/api_client.dart';
 import 'flavors.dart';
 
 void main() {
   runApp(SurveyApp());
 }
 
 class SurveyApp extends StatelessWidget {
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
+    _initAppDependencies();
```

Comment:
I think we should put `_initAppDependencies()` inside `main()` instead. The app only init dependencies one time, so putting it inside `build()` method is error-prone if the widget tree is accidentally rebuilt, right? 😄

Reply:
🤔  let me try

Reply:
the problem now is that we are having multiple main-staging/ main-production/main 🤔

Reply:
I'll switch to Stateful with @initState